src/trace/Kconfig

```diff
@@ -31,4 +31,59 @@ config TRACEM
 	help
 	  Sending all traces by mailbox additionally.
 
+config TRACE_FILTERING
+	bool "Trace filtering"
+	depends on TRACE
+	default y
+	help
+		Filtering of trace messages based on their verbosity level and velocity.
+
+config TRACE_FILTERING_VERBOSITY
+	bool "Filter by verbosity"
+	depends on TRACE_FILTERING
+	default y
+	help
+		Filtering by log verbosity level, where maximum verbosity allowed is specified for each
+		context and may be adjusted in runtime.
+
+config TRACE_FILTERING_ADAPTIVE
+	bool "Adaptive rate limiting"
+	depends on TRACE_FILTERING
+	default y
+	help
+		Adaptive filtering of trace messages, tracking up to CONFIG_TRACE_RECENT_ENTRIES_COUNT,
+		suppressing all repeated messages for up to CONFIG_TRACE_RECENT_TIME_THRESHOLD cycles.
+
+config TRACE_RECENT_ENTRIES_COUNT
+	int "Amount of entries considered recent"
+	depends on TRACE_FILTERING_ADAPTIVE
+	default 5
+	help
+		Recent entries are used to determine whether currently processed message was sent in the
+		past. Size of that filtering window affects effectiveness and performance. More recent
+		entries allow to better filter repetetive messeges out, but also slightly decrease
+		performance due to increased number of comparisions necessary.
+
+config TRACE_RECENT_TIME_THRESHOLD
+	int "Period of time considered recent in cycles"
+	depends on TRACE_FILTERING_ADAPTIVE
+	default 500000
+	help
+		Period of time during which entries are tracked and will be suppressed if reported again.
```

Comment:
What's the time unit here ?

Reply:
I think it's us, but I agree that it needs to be documented.

Reply:
> Just minor questions - one thing that does come up is that any user supplied trace filtering should disable the rate limiting at runtime (since the user is actively controlling trace). We need to make sure this will align.

@lgirdwood by user supplied trace filtering you are referring to additional filtering functionality that can be exposed at sof-logger application level?